SuppressionFiles doesn't supports interpolated strings

**Description:** _

> suppressionFiles when provided with interpolatedStrings gives java.lang.ArrayStoreException
_
**Example:**
suppressionFiles = [ "${rootProject.rootDir}/dependencyCheckSuppressionFile.xml",
                         "${rootProject.rootDir}/dependencyCheckSuppressionFile2.xml" ]

```
Caused by: java.lang.ArrayStoreException
        at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze.determineSuppressions(AbstractAnalyze.groovy:210)
        at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze$determineSuppressions.callCurrent(Unknown Source)
        at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze.initializeSettings(AbstractAnalyze.groovy:141)
        at org.owasp.dependencycheck.gradle.tasks.AbstractAnalyze.analyze(AbstractAnalyze.groovy:60)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:731)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:705)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 92 more

```

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ known, published vulnerabilities.
 ## Current Release
 The latest version is `2.1.1`.
 
-With the release of 2.1.1 the task name was changed from `depdencyCheck` to `dependencyCheckAnalyze`.
+With the release of 2.1.1 the task name was changed from `dependencyCheck` to `dependencyCheckAnalyze`.
 
 ## Usage
 Below are the quick start instructions. Please see the [documentation site](http://jeremylong.github.io/DependencyCheck/dependency-check-gradle/index.html)

--- a/README.md
+++ b/README.md
@@ -7,9 +7,9 @@ The dependency-check gradle plugin allows projects to monitor dependent librarie
 known, published vulnerabilities.
 
 ## Current Release
-The latest version is `2.0.0`.
+The latest version is `2.1.1`.
 
-With the release of 2.0.0 the task name was changed from `depdencyCheck` to `dependencyCheckAnalyze`.
+With the release of 2.1.1 the task name was changed from `depdencyCheck` to `dependencyCheckAnalyze`.
 
 ## Usage
 Below are the quick start instructions. Please see the [documentation site](http://jeremylong.github.io/DependencyCheck/dependency-check-gradle/index.html)
@@ -25,7 +25,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'org.owasp:dependency-check-gradle:2.0.0'
+        classpath 'org.owasp:dependency-check-gradle:2.1.1'
     }
 }
 
@@ -93,7 +93,7 @@ buildscript {
     mavenCentral()
   }
   dependencies {
-    classpath 'org.owasp:dependency-check-gradle:2.0.0'
+    classpath 'org.owasp:dependency-check-gradle:2.1.1'
   }
 }
 
@@ -110,7 +110,7 @@ buildscript {
     mavenCentral()
   }
   dependencies {
-    classpath 'org.owasp:dependency-check-gradle:2.0.0'
+    classpath 'org.owasp:dependency-check-gradle:2.1.1'
   }
 }
 

--- a/src/integTest/groovy/org/owasp/dependencycheck/gradle/DependencyCheckConfigurationSelectionIntegSpec.groovy
+++ b/src/integTest/groovy/org/owasp/dependencycheck/gradle/DependencyCheckConfigurationSelectionIntegSpec.groovy
@@ -29,7 +29,7 @@ class DependencyCheckConfigurationSelectionIntegSpec extends Specification {
         copyBuildFileIntoProjectDir('noSkipTestGroups.gradle')
 
         when:
-        def result = executeTaskAndGetResult(ANbuildALYZE_TASK, false)
+        def result = executeTaskAndGetResult(ANALYZE_TASK, false)
 
         then:
         result.task(":$ANALYZE_TASK").outcome == FAILED

--- a/src/integTest/groovy/org/owasp/dependencycheck/gradle/DependencyCheckConfigurationSelectionIntegSpec.groovy
+++ b/src/integTest/groovy/org/owasp/dependencycheck/gradle/DependencyCheckConfigurationSelectionIntegSpec.groovy
@@ -1,10 +1,12 @@
 package org.owasp.dependencycheck.gradle
 
+import org.gradle.testkit.runner.BuildResult
 import org.gradle.testkit.runner.GradleRunner
 import org.junit.Rule
 import org.junit.rules.TemporaryFolder
 import spock.lang.Specification
 import static org.gradle.testkit.runner.TaskOutcome.*
+import static org.owasp.dependencycheck.gradle.DependencyCheckPlugin.*
 
 class DependencyCheckConfigurationSelectionIntegSpec extends Specification {
 
@@ -130,5 +132,39 @@ class DependencyCheckConfigurationSelectionIntegSpec extends Specification {
         result.task(":$DependencyCheckPlugin.AGGREGATE_TASK").outcome == SUCCESS
         result.output.contains('CVE-2016-7051') //jackson cve from core
         result.output.contains('CVE-2015-6420') //commons cve from app
+    }
+
+    def "suppressionFiles argument can be parsed and files are being respected"() {
+        given:
+        copyBuildFileIntoProjectDir('suppressionFiles.gradle')
+        copyResourceFileIntoProjectDir('suppressions.xml', 'suppressions.xml')
+
+        when:
+        def result = executeTaskAndGetResult(ANALYZE_TASK, true)
+
+        then:
+        result.task(":$ANALYZE_TASK").outcome == SUCCESS
+    }
+
+
+    private void copyBuildFileIntoProjectDir(String buildFileName) {
+        copyResourceFileIntoProjectDir(buildFileName, 'build.gradle')
+    }
+
+    private void copyResourceFileIntoProjectDir(String resourceFileName, String targetFileName) {
+        def resourceFileContent = new File(getClass().getClassLoader().getResource(resourceFileName).toURI()).text
+        def targetDirectory = new File(testProjectDir.root, targetFileName).parentFile
+        targetDirectory.mkdirs()
+        def targetFile = testProjectDir.newFile(targetFileName)
+        targetFile << resourceFileContent
+    }
+
+    private BuildResult executeTaskAndGetResult(String taskName, boolean isBuildExpectedToPass) {
+        def build = GradleRunner.create()
+                .withProjectDir(testProjectDir.root)
+                .withArguments(taskName)
+                .withPluginClasspath()
+
+        isBuildExpectedToPass ? build.build() : build.buildAndFail()
     }
 }

--- a/src/integTest/groovy/org/owasp/dependencycheck/gradle/DependencyCheckConfigurationSelectionIntegSpec.groovy
+++ b/src/integTest/groovy/org/owasp/dependencycheck/gradle/DependencyCheckConfigurationSelectionIntegSpec.groovy
@@ -11,43 +11,28 @@ import static org.owasp.dependencycheck.gradle.DependencyCheckPlugin.*
 class DependencyCheckConfigurationSelectionIntegSpec extends Specification {
 
     @Rule final TemporaryFolder testProjectDir = new TemporaryFolder()
-    File buildFile
 
-    def setup() {
-        buildFile = testProjectDir.newFile('build.gradle')
-    }
 
     def 'test dependencies are ignored by default'() {
         given:
-        def resource = new File(getClass().getClassLoader().getResource('skipTestGroups.gradle').toURI())
-        buildFile << resource.text
+        copyBuildFileIntoProjectDir('skipTestGroups.gradle')
 
         when:
-        def result = GradleRunner.create()
-                .withProjectDir(testProjectDir.root)
-                .withArguments(DependencyCheckPlugin.ANALYZE_TASK)
-                .withPluginClasspath()
-                .withDebug(true)
-                .build()
+        def result = executeTaskAndGetResult(ANALYZE_TASK, true)
 
         then:
-        result.task(":$DependencyCheckPlugin.ANALYZE_TASK").outcome == SUCCESS
+        result.task(":$ANALYZE_TASK").outcome == SUCCESS
     }
 
     def "test dependencies are scanned if skipTestGroups flag is false"() {
         given:
-        def resource = new File(getClass().getClassLoader().getResource('noSkipTestGroups.gradle').toURI())
-        buildFile << resource.text
+        copyBuildFileIntoProjectDir('noSkipTestGroups.gradle')
 
         when:
-        def result = GradleRunner.create()
-                .withProjectDir(testProjectDir.root)
-                .withArguments(DependencyCheckPlugin.ANALYZE_TASK)
-                .withPluginClasspath()
-                .buildAndFail()
+        def result = executeTaskAndGetResult(ANbuildALYZE_TASK, false)
 
         then:
-        result.task(":$DependencyCheckPlugin.ANALYZE_TASK").outcome == FAILED
+        result.task(":$ANALYZE_TASK").outcome == FAILED
         result.output.contains('CVE-2015-6420')
         result.output.contains('CVE-2014-0114')
         result.output.contains('CVE-2016-3092')
@@ -56,80 +41,50 @@ class DependencyCheckConfigurationSelectionIntegSpec extends Specification {
 
     def "custom configurations are scanned by default"() {
         given:
-        def resource = new File(getClass().getClassLoader().getResource('scanCustomConfiguration.gradle').toURI())
-        buildFile << resource.text
+        copyBuildFileIntoProjectDir('scanCustomConfiguration.gradle')
 
         when:
-        def result = GradleRunner.create()
-                .withProjectDir(testProjectDir.root)
-                .withArguments(DependencyCheckPlugin.ANALYZE_TASK)
-                .withPluginClasspath()
-                .buildAndFail()
+        def result = executeTaskAndGetResult(ANALYZE_TASK, false)
 
         then:
-        result.task(":$DependencyCheckPlugin.ANALYZE_TASK").outcome == FAILED
+        result.task(":$ANALYZE_TASK").outcome == FAILED
         result.output.contains('CVE-2015-6420')
     }
 
     def "custom configurations are skipped if blacklisted"() {
         given:
-        def resource = new File(getClass().getClassLoader().getResource('blacklistCustomConfiguration.gradle').toURI())
-        buildFile << resource.text
+        copyBuildFileIntoProjectDir('blacklistCustomConfiguration.gradle')
 
         when:
-        def result = GradleRunner.create()
-                .withProjectDir(testProjectDir.root)
-                .withArguments(DependencyCheckPlugin.ANALYZE_TASK)
-                .withPluginClasspath()
-                .build()
+        def result = executeTaskAndGetResult(ANALYZE_TASK, true)
 
         then:
-        result.task(":$DependencyCheckPlugin.ANALYZE_TASK").outcome == SUCCESS
+        result.task(":$ANALYZE_TASK").outcome == SUCCESS
     }
 
     def "custom configurations are skipped when only scanning whitelisted configurations"() {
         given:
-        def resource = new File(getClass().getClassLoader().getResource('skipCustomConfigurationViaWhitelist.gradle').toURI())
-        buildFile << resource.text
+        copyBuildFileIntoProjectDir('skipCustomConfigurationViaWhitelist.gradle')
 
         when:
-        def result = GradleRunner.create()
-                .withProjectDir(testProjectDir.root)
-                .withArguments(DependencyCheckPlugin.ANALYZE_TASK)
-                .withPluginClasspath()
-                .build()
+        def result = executeTaskAndGetResult(ANALYZE_TASK, true)
 
         then:
-        result.task(":$DependencyCheckPlugin.ANALYZE_TASK").outcome == SUCCESS
+        result.task(":$ANALYZE_TASK").outcome == SUCCESS
     }
 
     def "aggregate task aggregates"() {
         given:
-        def resource = new File(getClass().getClassLoader().getResource('aggregateParent.gradle').toURI())
-        buildFile << resource.text
-
-        File settingsFile = testProjectDir.newFile('settings.gradle')
-        def settingsResource = new File(getClass().getClassLoader().getResource('aggregateSettings.gradle').toURI())
-        settingsFile << settingsResource.text
-
-        File appDir = testProjectDir.newFolder('app')
-        File app = new File(appDir,'build.gradle')
-        def appBuild = new File(getClass().getClassLoader().getResource('aggregateApp.gradle').toURI())
-        app << appBuild.text
-        File coreDir = testProjectDir.newFolder('core')
-        File core = new File(coreDir, 'build.gradle')
-        def coreBuild = new File(getClass().getClassLoader().getResource('aggregateCore.gradle').toURI())
-        core << coreBuild.text
+        copyBuildFileIntoProjectDir('aggregateParent.gradle')
+        copyResourceFileIntoProjectDir('aggregateSettings.gradle', 'settings.gradle')
+        copyResourceFileIntoProjectDir('aggregateApp.gradle', 'app/build.gradle')
+        copyResourceFileIntoProjectDir('aggregateCore.gradle', 'core/build.gradle')
 
         when:
-        def result = GradleRunner.create()
-                .withProjectDir(testProjectDir.root)
-                .withArguments(DependencyCheckPlugin.AGGREGATE_TASK)
-                .withPluginClasspath()
-                .build()
+        def result = executeTaskAndGetResult(AGGREGATE_TASK, true)
 
         then:
-        result.task(":$DependencyCheckPlugin.AGGREGATE_TASK").outcome == SUCCESS
+        result.task(":$AGGREGATE_TASK").outcome == SUCCESS
         result.output.contains('CVE-2016-7051') //jackson cve from core
         result.output.contains('CVE-2015-6420') //commons cve from app
     }

--- a/src/integTest/resources/suppressionFiles.gradle
+++ b/src/integTest/resources/suppressionFiles.gradle
@@ -1,0 +1,18 @@
+plugins {
+    id 'org.owasp.dependencycheck'
+}
+apply plugin: 'java'
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+}
+
+dependencies {
+    compile 'commons-collections:commons-collections:3.2'
+}
+
+dependencyCheck {
+    failBuildOnCVSS = 0
+    suppressionFiles = ["${project.rootDir}/suppressions.xml"]
+}

--- a/src/integTest/resources/suppressions.xml
+++ b/src/integTest/resources/suppressions.xml
@@ -1,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.1.xsd">
+    <suppress>
+        <notes><![CDATA[
+   file name: commons-collections-3.2.jar
+   ]]></notes>
+        <gav regex="true">^commons-collections:commons-collections:.*$</gav>
+        <cpe>cpe:/a:apache:commons_collections</cpe>
+    </suppress>
+</suppressions>

--- a/src/main/groovy/org/owasp/dependencycheck/gradle/tasks/AbstractAnalyze.groovy
+++ b/src/main/groovy/org/owasp/dependencycheck/gradle/tasks/AbstractAnalyze.groovy
@@ -203,11 +203,11 @@ abstract class AbstractAnalyze extends DefaultTask {
      *
      * @return an array of suppression file paths
      */
-    def determineSuppressions(suppressionFiles, suppressionFile) {
+    private String[] determineSuppressions(suppressionFiles, suppressionFile) {
         if (suppressionFile != null) {
-            suppressionFiles.add(suppressionFile)
+            suppressionFiles << suppressionFile
         }
-        return suppressionFiles.toArray(new String[0])
+        suppressionFiles
     }
     /**
      * Releases resources and removes temporary files used.

--- a/src/main/groovy/org/owasp/dependencycheck/gradle/tasks/Purge.groovy
+++ b/src/main/groovy/org/owasp/dependencycheck/gradle/tasks/Purge.groovy
@@ -60,7 +60,7 @@ class Purge extends DefaultTask {
                 }
             }
         } else {
-            String msg = "Unable to purge database; the database file does not exists: ${db.getAbsolutePath()}"
+            String msg = "Unable to purge database; the database file does not exist: ${db.getAbsolutePath()}"
             if (config.failOnError) {
                 throw new GradleException(msg)
             } else {

